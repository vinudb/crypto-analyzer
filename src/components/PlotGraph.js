import React from 'react';

import CanvasJSReact from '../libs/canvasjs.react';
var CanvasJSChart = CanvasJSReact.CanvasJSChart;

class PlotGraph extends React.Component {
    render() {
        const convertToBillion=(number)=> Math.abs(Number(number)) / 1.0e+9;

		const options = {
			animationEnabled: true,
			exportEnabled: true,
			theme: "light2", // "light1", "light2", "dark1", "dark2"
			title:{
				text: "Volume (24h) vs Abolute Price Change (24h) vs Market Capitalization",
			fontSize: 20
			},
			axisX: {
				title: "Market Cap (in Billion $)",
			logarithmic: true
			},
			axisY: {
				title: "Volume(24h) (in Billion $)"
			},
			data: [{
				type: "bubble",
				indexLabel: "",
				toolTipContent: "<b>{label}</b><br>Market Cap: ${x}bn <br>Volume (24h): ${y}bn<br>Price Change: {z}%",
                dataPoints: 
                    this.props.coins.map((coin)=>{
                        return(
                            {label: coin['name'], x: convertToBillion(coin['market_cap_usd']), y: convertToBillion(coin['24h_volume_usd']), z: Math.abs(coin.percent_change_24h)}
                        )
                    })
			}]
		}
		
		return (
			<CanvasJSChart options = {options} />
		);
	}
}
export default PlotGraph;