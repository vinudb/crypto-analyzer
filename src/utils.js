let _limit = 10;
export const setCurrentLimit = (limit)=>{
    _limit = limit;
};

export const getCurrentLimit = ()=>{
    return _limit;
}