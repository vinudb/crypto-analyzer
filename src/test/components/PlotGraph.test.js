import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PlotGraph from '../../components/PlotGraph';
import { mockCoinsResponse } from '../utils';
configure({ adapter: new Adapter() });

test('should render graph correctly', () => {
  const wrapper = shallow(<PlotGraph coins={mockCoinsResponse} />);
  expect(wrapper).toMatchSnapshot();
});
