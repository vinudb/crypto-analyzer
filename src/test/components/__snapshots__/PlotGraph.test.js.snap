// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render graph correctly 1`] = `ShallowWrapper {}`;
